packages/flutter/lib/src/services/restoration.dart

```diff
@@ -0,0 +1,984 @@
+/// serialize out its restoration data. When the user navigates back to the
+/// backgrounded app, it is restarted and the serialized restoration data is
+/// provided to it again. Ideally, the app will use that data to restore itself
+/// to the same state it was in when the user backgrounded the app.
+///
+/// In Flutter, restoration data is organized in a tree of [RestorationBucket]s
+/// which is rooted in the [rootBucket]. All information that the application
+/// needs to restore its current state must be stored in a bucket in this
+/// hierarchy. To store data in the hierarchy, entities (e.g. [Widget]s) must
+/// claim ownership of a child bucket from a parent bucket (which may be the
+/// [rootBucket] provided by this [RestorationManager]). The owner of a bucket
+/// may store arbitrary values in the bucket as long as they can be serialized
+/// with the [StandardMessageCodec]. The values are stored in the bucket under a
+/// given [RestorationId] as key. The [RestorationId]s used to store values must
+/// be unique within a given bucket. To access the stored value again during
+/// state restoration, the same [RestorationId] must be provided again. The
+/// owner of the bucket may also make the bucket available to other entities so
+/// that they can claim child buckets from it for their own restoration needs.
+/// Within a bucket, child buckets are also identified by unique
+/// [RestorationId]s. The restoration id must be provided when claiming a child
+/// bucket.
+///
+/// When restoration data is provided to the [RestorationManager] (e.g. after
+/// the application relaunched when foregrounded again), the bucket hierarchy
+/// with all the data stored in it is restored. Entities can retrieve the data
+/// again by using the same restoration ids that they originally used to store
+/// the data.
+///
+/// In addition to providing restoration data when the app is launched,
+/// restoration data may also be provided to a running app to restore it to a
+/// previous state. When this happens, the current bucket hierarchy is
+/// decommissioned and replaced with the hierarchy deserialized from the newly
+/// provided restoration data. Buckets in the old hierarchy notify their
+/// listeners when they get decommissioned. In response to the notification,
+/// listeners must stop using the old buckets. Owners of those buckets must
+/// dispose of them and claim a new child as a replacement from a parent in the
+/// new bucket hierarchy (that parent may be the updated [rootBucket]).
+///
+/// Same platforms restrict the size of the restoration data. Therefore, the
+/// data stored in the buckets should be as small as possible while still
+/// allowing the app to restore its current state from it. Data that can be
+/// retrieved from other services (e.g. a database or a web server) should not
+/// be included in the restoration data. Instead, a small identifier (e.g. an ID
```

Comment:
ID stands for identifier, so "Instead, a small identifier (e.g. an identifier or resource locator)" is weird. Maybe "e.g. a UUID" or "e.g. a database record number" or something?